### PR TITLE
Removing setFocus from components

### PR DESCRIPTION
**Pull request by gretanausedaite**
## Changes

Most of components have refs. Therefore custom setFocus functionality is not needed anymore - users can set focus themselves using said ref.

Removed `setFocus` from:
- Checkbox
- ColorPicker
- ComboBox
- Input
- LabeledInput
- LabeledTextarea
- Radio
- RadioTile
- Select
- Slider
- ToggleSwitch

Added setFocus functionality to:
- TextFilter
- NumberRangeFilter
- DateRangeFilter

Keeping setFocus on (because of more complex use case):
- DatePicker
- Dialog
- Menu
- TimePicker

## Testing

- Updated unit tests to manually focus 
- Visual tests pass

## Docs

- [x] changeset
- [ ] wiki

### CODE REVIEW

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx [hunk 1]
```diff
@@ -34,6 +39,7 @@ const DatePickerInput = (props: DatePickerInputProps) => {
     formatDate,
     isFromOrTo,
     selectedDate,
+    setFocus = false,
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx:42]: cant we forward the ref instead?

**Review by mayank99 — APPROVED** [review]
Nice work!

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx:42]: yeah, i'm ok with keeping all the setFocus stuff in `-Filter` components for now.

Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx]: Moved to DateRangeFilter

File: packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx [hunk 2]
```diff
@@ -3,7 +3,12 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
-import { Popover, SvgCalendar, isBefore } from '../../../utils/index.js';
+import {
+  Popover,
+  SvgCalendar,
+  isBefore,
+  type PolymorphicForwardRefComponent,
```
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Table/filters/DateRangeFilter/DatePickerInput.tsx]: updated. Why VS code is not smart enough to separate type imports?

**Review by mayank99 — APPROVED** [review]

**Pull request merged**